jenkins/PerfCI_Chaos/03_PerfCI_Choas_VMs_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,191 @@
+        KUBECONFIG_PATH = '/root/.kube/config'
+        PRIVATE_KEY_PATH = '/home/jenkins/.ssh/provision_private_key'
+        CONFIG_PATH = '/home/jenkins/.ssh/config'
+        WORKSPACE='/home/jenkins'
+        SAVE_ARTIFACTS_LOCAL = 'False'
+        ENABLE_PROMETHEUS_SNAPSHOT = 'True'
+        DELETE_ALL = 'False' // Not delete the running VMs
+        RUN_STRATEGY = 'True' // runStrategy: Always for NHC/ SNR operators
+        RUN_TYPE = 'chaos_ci'
+        BOOTSTORM_SCALE = 80
+        THREADS_LIMIT= 20
+        TIMEOUT = 8000
+        WORKSPACE_PATH = "${env.WORKSPACE}/workspace/${env.JOB_NAME}/"
+    }
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+                        sh '''
+                            # Check if the image exists
+                            if [[ "$(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)" != "" ]]; then
+
+                                # Get containers using the image
+                                containers=$(sudo podman ps -a --filter ancestor=${QUAY_BENCHMARK_RUNNER_REPOSITORY} -q)
+
+                                if [[ -n "$containers" ]] ; then
+                                    echo "Stopping and removing containers using the image..."
+                                    sudo podman stop $containers
+                                    sudo podman rm -f $containers
+                                fi
+
+                                # Now remove the image
+                                sudo podman rmi -f $(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)
+                            fi
+                        '''
+                }
+            }
+        }
+
+        stage('⚙️ SET SSH key') {
+            steps {
+                script {
+                    sh 'umask 700 && mkdir -p "$WORKSPACE"/.ssh'
+                    withCredentials([file(credentialsId: 'perfci_provision_private_key_file', variable: 'PROVISION_PRIVATE_KEY_FILE')]) {
+                        sh "sudo cp ${PROVISION_PRIVATE_KEY_FILE} "$WORKSPACE"/.ssh/provision_private_key"
+                        sh "sudo chown jenkins:jenkins "$WORKSPACE"/.ssh/provision_private_key"
+                    }
+                    sh 'sudo chmod 600 $PRIVATE_KEY_PATH'
+                    sh '''
+                        sudo cat > "$CONFIG_PATH" <<END
+    Host provision
+        HostName ${PROVISION_IP}
+        User ${PROVISION_USER}
+        IdentityFile $PRIVATE_KEY_PATH
+        StrictHostKeyChecking no
+        ServerAliveInterval 30
+        ServerAliveCountMax 5
+END
+                    '''
+                    sh 'sudo chmod 600 $CONFIG_PATH'
+                }
+            }
+        }
+
+        stage('Prepare Google Credential Files') {
+            steps {
+                script {
```

Comment:
I assume it's not really bash running this?

Reply:
Yes, its a groovy code.